test/test_extended_models.py

```diff
@@ -59,6 +60,24 @@ def test_get_model_weights(name, weight):
     assert models.get_model_weights(name) == weight
 
 
+@pytest.mark.parametrize("copy_fn", [copy.copy, copy.deepcopy])
+@pytest.mark.parametrize(
+    "name",
+    [
+        "resnet50",
+        "retinanet_resnet50_fpn_v2",
+        "raft_large",
+        "quantized_resnet50",
+        "lraspp_mobilenet_v3_large",
+        "mvit_v1_b",
+    ],
+)
+def test_weights_copyable(copy_fn, name):
+    weights = models.get_model_weights(name)
+    copied_weights = copy_fn(weights)
+    assert copied_weights is weights
```

Comment:
So that's equivalent to just doing `copied_weights = weights`; this isn't what we would expect of a `copy` or a `deepcopy`, right?

Reply:
Correct, that is certainly not what I would expect. Buuut, this seems to be the normal behavior of the builtin `enum.Enum`, so I think we can just align our behavior to that. The [documentation](The most interesting thing about enum members is that they are singletons. [EnumType](https://docs.python.org/3/library/enum.html#enum.EnumType) creates them all while it is creating the enum class itself, and then puts a custom __new__() in place to ensure that no new ones are ever instantiated by returning only the existing member instances.) states

> The most interesting thing about enum members is that they are singletons. `EnumType` creates them all while it is creating the enum class itself, and then puts a custom `__new__()` in place to ensure that no new ones are ever instantiated by returning only the existing member instances.

Thus, (deep-)copying is not possible by design, because then you would have two objects and in turn violate the singleton rule.